src/aggregator/aggregator/generic_elem.go

```diff
@@ -433,13 +452,16 @@ func (e *GenericElem) Consume(
 		e.values[dirtyTime] = val
 		e.toConsume = append(e.toConsume, flushState)
 
-		// potentially consume the nextAgg as well in case we need to cascade an update due to a resend.
-		nextAgg, ok := e.nextAggWithLock(dirtyTime, targetNanos, isEarlierThanFn)
-		if ok && (i == len(dirtyTimes)-1 || dirtyTimes[i+1] != nextAgg.startAtNanos) {
-			// if not already in the dirty set.
-			flushState, ready := e.updateFlushStateWithLock(nextAgg.startAtNanos, targetNanos, isEarlierThanFn)
-			if ready {
-				e.toConsume = append(e.toConsume, flushState)
+		// potentially consume the nextAgg as well in case we need to cascade an update from a previously flushed
+		// value.
+		if flushState.flushed {
```

Comment:
figured this guard would help reduce some unnecessary lookups in the common case.

Reply:
So this can be done since a "nextAgg" can only precede a "flushed current agg"? Might be worth commenting

Reply:
it's more we only care to cascade an update when a previously flushed value has been updated. will add a comment/example.